src/test/java/org/opensearch/sdk/TestNamedWriteableRegistryApi.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.sdk;
+
+import static java.util.Collections.emptySet;
+import static java.util.Collections.emptyMap;
+
+import org.opensearch.Version;
+import org.opensearch.cluster.node.DiscoveryNode;
+import org.opensearch.common.io.stream.NamedWriteable;
+import org.opensearch.common.io.stream.NamedWriteableRegistry;
+import org.opensearch.common.io.stream.NamedWriteableRegistryParseRequest;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.NamedWriteableRegistryResponse;
+import org.opensearch.common.io.stream.OutputStreamStreamOutput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.transport.TransportAddress;
+import org.opensearch.test.OpenSearchTestCase;
+import org.opensearch.extensions.BooleanResponse;
+import org.opensearch.extensions.OpenSearchRequest;
+import org.opensearch.extensions.ExtensionsOrchestrator.OpenSearchRequestType;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class TestNamedWriteableRegistryApi extends OpenSearchTestCase {
+    private List<NamedWriteableRegistry.Entry> namedWriteables;
+    private NamedWriteableRegistryApi namedWriteableRegistryApi;
+
+    private static class Example implements NamedWriteable {
```

Comment:
Why are we creating this class here? Why not just create an object of NamedWriteableRegistryAPI and test it?

Reply:
The class is an example of a writeable reader that is registered within the NamedWriteableRegistry of the API. This is used to test deserialization and registration